vpr/src/place/initial_placement.cpp

```diff
@@ -963,6 +977,16 @@ static void place_all_blocks(vtr::vector<ClusterBlockId, t_block_score>& block_s
             heap_blocks.pop_back();
 
             auto blk_id_type = cluster_ctx.clb_nlist.block_type(blk_id);
+
+            const auto& cluster_blk_pb_type = cluster_ctx.clb_nlist.block_type(blk_id)->pb_type;
```

Comment:
1. This should be a subroutine I think (cleaner, less cognitive pollution for the main code path).
2. It should be guarded by your #ifdef VTR_ENABLE_DEBUG_LOGGING